packages/lodestar/src/db/single/preGenesisState.ts

```diff
@@ -2,26 +2,30 @@ import {TreeBacked, ContainerType} from "@chainsafe/ssz";
 import {GENESIS_SLOT} from "@chainsafe/lodestar-params";
 import {allForks} from "@chainsafe/lodestar-types";
 import {IBeaconConfig} from "@chainsafe/lodestar-config";
-import {IDatabaseController, Bucket} from "@chainsafe/lodestar-db";
+import {IDatabaseController, Bucket, IDbMetrics} from "@chainsafe/lodestar-db";
 
 export class PreGenesisState {
   private readonly config: IBeaconConfig;
   private readonly bucket: Bucket;
   private readonly db: IDatabaseController<Buffer, Buffer>;
   private readonly key: Buffer;
+  private readonly metrics?: IDbMetrics;
 
-  constructor(config: IBeaconConfig, db: IDatabaseController<Buffer, Buffer>) {
+  constructor(config: IBeaconConfig, db: IDatabaseController<Buffer, Buffer>, metrics?: IDbMetrics) {
     this.config = config;
     this.db = db;
-    this.bucket = Bucket.phase0_preGenesisState as Bucket;
+    this.bucket = Bucket.phase0_preGenesisState;
     this.key = Buffer.from(new Uint8Array([this.bucket]));
+    this.metrics = metrics;
   }
 
   async put(value: TreeBacked<allForks.BeaconState>): Promise<void> {
+    this.metrics?.dbWrites.labels({bucket: "phase0_preGenesisState"}).inc();
```

Comment:
Why is it necessary to manually call the metrics here? Shouldn't `this.db` call it on the Abstract class?

Reply:
this repo is not extending AbstractRepository. this.db is controller, not repository